src/System.CommandLine.Hosting/InvocationLifetime.cs

```diff
@@ -17,7 +17,7 @@
 
 namespace System.CommandLine.Hosting
 {
-    public class InvocationLifetime : IHostLifetime
+    public sealed class InvocationLifetime : IHostLifetime
```

Comment:
What is  the reason for this change?

Reply:
From this issue https://github.com/dotnet/runtime/issues/49944, sealed types would get performance gains and there're no virtual methods to override so think it could be a sealed class, same for the `InvocationLifetimeOptions`

Reply:
Reverted

Reply:
This is good performance advice but generally something I'd avoid applying universally without a benchmark indicating that these specific types will benefit from being `sealed`.